selfdrive/ui/qt/widgets/cameraview.cc

```diff
@@ -209,13 +224,17 @@ void CameraViewWidget::paintGL() {
   }
 
   glBindVertexArray(frame_vao);
-  glActiveTexture(GL_TEXTURE0);
-  glBindTexture(GL_TEXTURE_2D, texture[latest_texture_id]->frame_tex);
 
   glUseProgram(program->programId());
-  glUniform1i(program->uniformLocation("uTexture"), 0);
-  glUniformMatrix4fv(program->uniformLocation("uTransform"), 1, GL_TRUE, frame_mat.v);
+  for (int i = 0; i < 3; ++i) {
```

Comment:
I don't understand why this loop has to be here at all.

Reply:
It tells the shader which texture unit points to which which texture buffer